### ISSUE TITLE
Importing archive with group that already exists excepts

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#2074]
When attempting to import an archive that contains a `Group` with a name that already exists, the following exception will be thrown:
```
Traceback (most recent call last):
  File "/home/sphuber/code/aiida/env/dev/aiida-core/aiida/cmdline/commands/cmd_import.py", line 72, in cmd_import
    import_data(archive)
  File "/home/sphuber/code/aiida/env/dev/aiida-core/aiida/orm/importexport.py", line 320, in import_data
    silent=silent)
  File "/home/sphuber/code/aiida/env/dev/aiida-core/aiida/orm/importexport.py", line 1256, in import_data_sqla
    session.flush()
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2112, in flush
    self._flush(objects)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2230, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2194, in _flush
    flush_context.execute()
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 373, in execute
    rec.execute(self)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 532, in execute
    uow
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 178, in save_obj
    mapper, table, insert)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 804, in _emit_insert_statements
    execute(statement, params)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
    exc_info
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/home/sphuber/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
IntegrityError: (psycopg2.IntegrityError) duplicate key value violates unique constraint "db_dbgroup_name_type_key"
DETAIL:  Key (name, type)=(cod_structure_primitive, ) already exists.
 [SQL: 'INSERT INTO db_dbgroup (uuid, name, type, time, description, user_id) VALUES (%(uuid)s, %(name)s, %(type)s, %(time)s, %(description)s, %(user_id)s) RETURNING db_dbgroup.id'] [parameters: {'user_id': 2, 'description': u'', 'name': u'cod_structure_primitive', 'time': datetime.datetime(2018, 10, 17, 7, 25, 34, 977405, tzinfo=<UTC>), 'type': u'', 'uuid': u'ec7aacc6-1b23-4a30-8805-47cb5b2c3a7c'}]

do you want to continue? [y/N]: N
```
This should be checked for and handled more gracefully

**Comment by giovannipizzi** [comment]
If we want to keep the uniqueness constraint, we should deduplicate the groups like we do now for Computer names (however, we should be careful: if we import twice the same file, we should NOT keep deduplicating it! Use the group UUID for this).

**Comment by giovannipizzi** [comment]
Not sure, I think we should allow for importing and automatically append something at the end of the name (like `- 1`, `- 2`, ... - the smallest integer that makes the group name unique).

**Comment by lekah** [comment]
The behavior I would want is for the importer to recognize that this is the same group as the one existing in the DB and adding nodes to the existing group instead of adding a new group.
In this way 2 researchers A and B could sync their groups via export files...
This is currently quite impossible due to the complicated implementation. 
The comment by @giovannipizzi is a good compromise for now, given uuids are different...

**Comment by sphuber** [comment]
The names of two groups being identical does not mean that they are the same semantically. Therefore I think automatic merging is dangerous

**Comment by lekah** [comment]
Ok,  so for now I will  update the names. Same uuids is handled elsewhere.

**Comment by sphuber** [comment]
Fixed in PR #2270
